src/shared/tpm2-util.c

```diff
@@ -1304,7 +1306,9 @@ int tpm2_make_luks2_json(
                                        JSON_BUILD_PAIR("tpm2-pcrs", JSON_BUILD_VARIANT(a)),
                                        JSON_BUILD_PAIR_CONDITION(!!tpm2_pcr_bank_to_string(pcr_bank), "tpm2-pcr-bank", JSON_BUILD_STRING(tpm2_pcr_bank_to_string(pcr_bank))),
                                        JSON_BUILD_PAIR_CONDITION(!!tpm2_primary_alg_to_string(primary_alg), "tpm2-primary-alg", JSON_BUILD_STRING(tpm2_primary_alg_to_string(primary_alg))),
-                                       JSON_BUILD_PAIR("tpm2-policy-hash", JSON_BUILD_HEX(policy_hash, policy_hash_size))));
+                                       JSON_BUILD_PAIR("tpm2-policy-hash", JSON_BUILD_HEX(policy_hash, policy_hash_size)),
+                                       JSON_BUILD_PAIR_CONDITION(use_pin, "tpm2-flags", JSON_BUILD_ARRAY(JSON_BUILD_STRING("pin"))))
```

Comment:
hmm, i'd just serialization that as bool. i.e.

```c
JSON_BUILD_PAIR_CONDITION(use_pin, "tpm2-flags", JSON_BUILD_BOOLEAN(true)),
```

i.e. if we have high-level, friendly bools anyway no need to group thos together with likely unrelated flags.

Reply:
Hm, my idea is that we could have a generic list of flags that would influence the TPM policy. If you don't like it that's OK. I was thinking of issue #20668 for instance. We could add a new flag keyword to signal that the primary key should be made with DA protection disabled. If I make this a top-level flag instead I should probably rename it, too.